theano/sandbox/cuda/opt.py

```diff
@@ -441,6 +441,24 @@ def local_gpu_dot_to_dot22(node):
     return False
 
 
+@local_optimizer([gpu_from_host, host_from_gpu])
+def local_assert_no_cpu_op(node):
+    if not isinstance(node.op, GpuOp) and all([var.owner and isinstance(var.owner.op,
+        HostFromGpu) for var in node.inputs]) and all([var.owner and
+            isinstance(var.owner.op, GpuFromHost) for var in node.outputs]):
+            if config.assert_no_cpu_op == "warn":
+                _logger.warning(("CPU op %s is detected in the computational"
+                                 " graph") % node)
+            elif config.assert_no_cpu_op == "raise":
+                raise RuntimeError("The op %s is on CPU." % node)
+            elif config.assert_no_cpu_op == "pdb":
+                import ipdb; ipdb.set_trace()
+    return None
+
+if config.assert_no_cpu_op != "ignore" and config.assert_no_cpu_op:
```

Comment:
Because the default value is ignore, I didn't want to register if it is ignore or assert_no_cpu_op is not specified. But I am not sure how much would the overhead of this will be.